clients/python-apiserver-client/python_apiserver_client/params/jobsubmission.py

```diff
@@ -0,0 +1,108 @@
+import datetime
+
+
+class RayJobRequest:
+    """
+    RayJobRequest used to define job to be submitted to a Ray cluster
+    It provides APIs to create, stringify and convert to dict.
+
+    Methods:
+    - Create RayJobRequest: gets the following parameters:
+        entrypoint - required, the command to start a job on the cluster
+        submission_id - optional, submission id for the job submission
+        runtime_env - optional, yaml string specifying job runtime environment
+        metadata - optional, dictionary of the submission metadata
+        num_cpus - optional, number of cpus for job execution
+        num_gpus - optional, number of gpus for job execution
+        resources - optional, dictionary of the resources for job execution
+    """
+    def __init__(self, entrypoint: str, submission_id: str = None, runtime_env: str = None,
+                 metadata: dict[str, str] = None, num_cpu: float = -1., num_gpu: float = -1.,
```

Comment:
Nit: Why use -1 here? In the Ray Job API, if these fields are not specified, it defaults to 0.

Reply:
I did not want to print/include in dict things that are not defined